### PR TITLE
[MM-45] Update plugin with respect to phase 1 upgrades

### PR DESCRIPTION
**Pull request by ayusht2810**
#### Summary
Updated plugin with respect to phase 1 upgrades

- Updated Makefile
- Updated version in `nvmrc` file and update `package-lock.json` file.
- Updated `node-sass` to `sass`
- Updated developer docs link in the readme file 
- Updated package in go modules to use https://pkg.go.dev/github.com/mattermost/mattermost/server/public.

#### Ticket Link

### CODE REVIEW

**Review by mickmister — APPROVED** [review]
Awesome work @ayusht2810!

**Review by AayushChaudhary0001 — APPROVED** [review]
Tested for server and cloud both, working fine for all the basic flows. Approved.

File: readme.md [hunk 1]
```diff
@@ -422,7 +422,16 @@ This repository is licensed under the Apache 2.0 License, except for the [server
...
+This plugin contains both a server and web app portion. Read our documentation about the [Developer Workflow](https://developers.mattermost.com/integrate/plugins/developer-workflow/) and [Developer Setup](https://developers.mattermost.com/integrate/plugins/developer-setup/) for more information about developing and extending plugins.
+
+### Releasing new versions
+
+The version of a plugin is determined at compile time, automatically populating a `version` field in the [plugin manifest](plugin.json):
+* If the current commit matches a tag, the version will match after stripping any leading `v`, e.g. `1.3.1`.
+* Otherwise, the version will combine the nearest tag with `git rev-parse --short HEAD`, e.g. `1.3.1+d06e53e1`.
+* If there is no version tag, an empty version will be combined with the short hash, e.g. `0.0.0+76081421`.
+
+To disable this behaviour, manually populate and maintain the `version` field.
```
Comment (mickmister) [inline readme.md:434]: Wondering how version bumps work in this case. Would we just not have version bump PRs anymore? The tag is what decides the version now IIUC. @hanzei Thoughts on this? We typically use version bump PRs as a means to publicly discuss things about the releases

File: webapp/src/actions/index.ts [hunk 2]
```diff
@@ -434,7 +434,7 @@ export function handleConnectFlow(instanceID?: string) {
 export function redirectConnect(instanceID: string) {
     return async (dispatch, getState) => {
         const instancePrefix = '/instance/' + btoa(instanceID);
-        const target = '/plugins/' + PluginId + instancePrefix + '/user/connect';
+        const target = '/plugins/' + manifest.id + instancePrefix + '/user/connect';
```
Comment (mickmister) [inline webapp/src/actions/index.ts:437]: Should be using SiteURL here

**Review by mickmister — APPROVED** [review]

File: readme.md (hunk 1 above)
Comment (hanzei) [reply readme.md:434]: @mickmister I like version bump PRs to as a sync point for all parties involved. Hence, I'm leaning towards keeping the process as it is for major plugins like JIRA.

**Pull request merged**